syntax = "proto3";
package state;
import "validate/validate.proto";
import "meta/meta.proto";
import "substate.proto";

message State {
  optional StateInternal internal = 1;
  optional substate.State inherited = 2;
  optional uint32 global_param = 3;

}

message StateInternal {
  uint32 param1 = 1 [(validate.rules).uint32 = {lte: 10}, (meta.meta) = {name: "hello"}];
}